--- gtetrinet-GTETRINET_0_7_11/configure.ac	2016-08-19 03:34:38.621646261 +0100
+++ gtetrinet-GTETRINET_0_7_11.new/configure.ac	2016-08-19 03:43:46.894830573 +0100
@@ -27,9 +27,6 @@ AC_CHECK_HEADERS([ctype.h dirent.h errno
 AC_CHECK_FUNCS([memset setlocale])
 
 # Initialise i18n
-AM_GNU_GETTEXT_VERSION([0.17])
-AM_GNU_GETTEXT([external])
-
 GETTEXT_PACKAGE=gtetrinet
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [Define to the gettext domain name.])
