diff --git a/configure.ac b/configure.ac
index 929ee53..8711c97 100644
--- a/configure.ac
+++ b/configure.ac
@@ -49,32 +49,34 @@ if test -d $srcdir/.git; then
 fi
 
 # Check for required packages
-LIBGTK_REQUIRED=2.6.0
+LIBGLIB_REQUIRED=2.32.0
+LIBGTK_REQUIRED=2.18.0
 LIBGNOME_REQUIRED=2.0.0
 LIBGNOMEUI_REQUIRED=2.0.0
-LIBESD_OPTIONAL=0.2.36
+LIBCANBERRAGTK_OPTIONAL=0.2
 
 PKG_CHECK_MODULES(GTETRINET,[
+                  glib-2.0 >= $LIBGLIB_REQUIRED
                   gtk+-2.0 >= $LIBGTK_REQUIRED
                   libgnome-2.0 >= $LIBGNOME_REQUIRED
                   libgnomeui-2.0 >= $LIBGNOMEUI_REQUIRED
 ])
 
-# Check for libesd
-if pkg-config --exists esound ; then
-	have_esound=yes
-else 
-	have_esound=no
+# Check for libcanberra-gtk
+if pkg-config --exists "libcanberra-gtk >= $LIBCANBERRAGTK_OPTIONAL"; then
+	have_canberragtk=yes
+else
+	have_canberragtk=no
 fi
 
-if test "x$have_esound" = "xyes"; then
-	AC_DEFINE([HAVE_ESD],[1],[Define this to enable EsounD support.])
-	PKG_CHECK_MODULES([ESOUND],[esound >= $LIBESD_OPTIONAL])
-	AC_SUBST(ESOUND_CFLAGS)
-	AC_SUBST(ESOUND_LIBS)
+if test "x$have_canberragtk" = "xyes"; then
+	AC_DEFINE([HAVE_CANBERRAGTK],[1],[Define this to enable sound support.])
+	PKG_CHECK_MODULES([CANBERRAGTK],[libcanberra-gtk >= $LIBCANBERRAGTK_OPTIONAL])
+	AC_SUBST(CANBERRAGTK_CFLAGS)
+	AC_SUBST(CANBERRAGTK_LIBS)
 fi
 
-AM_CONDITIONAL(HAVE_ESOUND, test "x$have_esound" = "xyes")
+AM_CONDITIONAL(HAVE_CANBERRAGTK, test "x$have_canberragtk" = "xyes")
 
 AC_CHECK_FUNC(putenv)
 AC_CHECK_FUNC(setenv)
@@ -156,5 +158,5 @@ echo "   GTranslator version: ${PACKAGE_VERSION}"
 echo "   Compilation flags:   ${CFLAGS}"
 echo "   CPPFLAGS:            ${CPPFLAGS}"
 echo "   IPv6 support:        $enable_ipv6"
-echo "   EsounD support:      $have_esound"
+echo "   Sound support:       $have_canberragtk"
 echo "   Page detaching:      $enable_detach"
diff --git a/src/Makefile.am b/src/Makefile.am
index e750b18..71cc072 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -4,8 +4,8 @@ localedir = $(datadir)/locale
 
 AM_CFLAGS = $(GTETRINET_CFLAGS)
 
-if HAVE_ESOUND
-gtetrinet_LDADD = $(GTETRINET_LIBS) $(ESOUND_LIBS)
+if HAVE_CANBERRAGTK
+gtetrinet_LDADD = $(GTETRINET_LIBS) $(CANBERRAGTK_LIBS)
 else
 gtetrinet_LDADD = $(GTETRINET_LIBS)
 endif
diff --git a/src/client.c b/src/client.c
index 7c6d645..2dbce3e 100644
--- a/src/client.c
+++ b/src/client.c
@@ -33,7 +33,7 @@
 #include <errno.h>
 #include <signal.h>
 #include <ctype.h>
-#include <gnome.h>
+#include <glib/gi18n.h>
 
 #include "client.h"
 #include "tetrinet.h"
@@ -225,7 +225,7 @@ void client_process (void)
   errno = 0;
   resolved = 0;
   
-  thread = g_thread_create ((GThreadFunc) client_resolv_hostname, NULL, FALSE, NULL);
+  thread = g_thread_new ("resolve", (GThreadFunc) client_resolv_hostname, NULL);
   
   /* wait until the hostname is resolved */
   while (resolved == 0)
@@ -234,6 +234,8 @@ void client_process (void)
       gtk_main_iteration ();
   }
 
+  g_thread_join (thread);
+
   if (resolved == -1) {
     char errmsg[1024];
 
@@ -294,7 +296,6 @@ gpointer client_resolv_hostname (void)
 #ifdef ENABLE_IPV6
     char hbuf[NI_MAXHOST];
     struct addrinfo hints, *res, *res0;
-    struct sockaddr_in6 sa;
     char service[10];
 #else
     struct hostent *h;
@@ -504,7 +505,7 @@ void server_ip (unsigned char buf[4])
 #else
     struct sockaddr_in sin;
 #endif
-    int len = sizeof(sin);
+    socklen_t len = sizeof(sin);
 
     getpeername (sock, (struct sockaddr *)&sin, &len);
 #ifdef ENABLE_IPV6
diff --git a/src/commands.c b/src/commands.c
index 7588acf..fc5f654 100644
--- a/src/commands.c
+++ b/src/commands.c
@@ -34,12 +34,9 @@
 #include "commands.h"
 #include "dialogs.h"
 
-#include "images/play.xpm"
-#include "images/pause.xpm"
-#include "images/stop.xpm"
 #include "images/team24.xpm"
-#include "images/connect.xpm"
-#include "images/disconnect.xpm"
+
+#define GTET_STOCK_TEAM24 "gtet-team24"
 
 GnomeUIInfo gamemenu[] = {
     GNOMEUIINFO_ITEM(N_("_Connect to server..."), NULL, connect_command, NULL),
@@ -79,23 +76,36 @@ GnomeUIInfo menubar[] = {
 };
 
 GnomeUIInfo toolbar[] = {
-    GNOMEUIINFO_ITEM_DATA(N_("Connect"), N_("Connect to a server"), connect_command, NULL, connect_xpm),
-    GNOMEUIINFO_ITEM_DATA(N_("Disconnect"), N_("Disconnect from the current server"), disconnect_command, NULL, disconnect_xpm),
+    GNOMEUIINFO_ITEM_STOCK(N_("Connect"), N_("Connect to a server"), connect_command, GTK_STOCK_CONNECT),
+    GNOMEUIINFO_ITEM_STOCK(N_("Disconnect"), N_("Disconnect from the current server"), disconnect_command, GTK_STOCK_DISCONNECT),
     GNOMEUIINFO_SEPARATOR,
-    GNOMEUIINFO_ITEM_DATA(N_("Start game"), N_("Start a new game"), start_command, NULL, play_xpm),
-    GNOMEUIINFO_ITEM_DATA(N_("End game"), N_("End the current game"), end_command, NULL, stop_xpm),
-    GNOMEUIINFO_ITEM_DATA(N_("Pause game"), N_("Pause the game"), pause_command, NULL, pause_xpm),
+    GNOMEUIINFO_ITEM_STOCK(N_("Start game"), N_("Start a new game"), start_command, GTK_STOCK_MEDIA_PLAY),
+    GNOMEUIINFO_ITEM_STOCK(N_("End game"), N_("End the current game"), end_command, GTK_STOCK_MEDIA_STOP),
+    GNOMEUIINFO_ITEM_STOCK(N_("Pause game"), N_("Pause the game"), pause_command, GTK_STOCK_MEDIA_PAUSE),
     GNOMEUIINFO_SEPARATOR,
-    GNOMEUIINFO_ITEM_DATA(N_("Change team"), N_("Change your current team name"), team_command, NULL, team24_xpm),
+    GNOMEUIINFO_ITEM_STOCK(N_("Change team"), N_("Change your current team name"), team_command, GTET_STOCK_TEAM24),
 #ifdef ENABLE_DETACH
     GNOMEUIINFO_SEPARATOR,
-    GNOMEUIINFO_ITEM_STOCK(N_("Detach page"), N_("Detach the current notebook page"), detach_command, "gtk-cut"),
+    GNOMEUIINFO_ITEM_STOCK(N_("Detach page"), N_("Detach the current notebook page"), detach_command, GTK_STOCK_CUT),
 #endif
     GNOMEUIINFO_END
 };
 
 void make_menus (GnomeApp *app)
 {
+  GtkIconFactory *icon_factory;
+  GdkPixbuf *team24_pixbuf;
+  GtkIconSet *team24_icon_set;
+
+  icon_factory = gtk_icon_factory_new ();
+  team24_pixbuf = gdk_pixbuf_new_from_xpm_data ((const char **)team24_xpm);
+  team24_icon_set = gtk_icon_set_new_from_pixbuf (team24_pixbuf);
+  g_object_unref (team24_pixbuf);
+  gtk_icon_factory_add (icon_factory, GTET_STOCK_TEAM24, team24_icon_set);
+  gtk_icon_set_unref (team24_icon_set);
+  gtk_icon_factory_add_default (icon_factory);
+  g_object_unref (icon_factory);
+
   gnome_app_create_menus (app, menubar);
 
   gtk_accel_map_add_entry ("<GTetrinet-Main>/Game/Start", gdk_keyval_from_name ("n"), GDK_CONTROL_MASK);
@@ -265,35 +275,6 @@ void commands_checkstate ()
     else partyline_status (_("Not connected"));
 }
 
-enum {
-	LINK_TYPE_EMAIL,
-	LINK_TYPE_URL
-};
-/* handle the links */
-
-void handle_links (GtkAboutDialog *about G_GNUC_UNUSED, const gchar *link, gpointer data)
-{
-    gchar *new_link;
-
-    switch (GPOINTER_TO_INT (data)) {
-    case LINK_TYPE_EMAIL:
-	    new_link = g_strdup_printf ("mailto: %s", link);
-	    break;
-    case LINK_TYPE_URL:
-	    new_link = g_strdup (link);
-	    break;
-    default:
-	    g_assert_not_reached ();
-    }
-
-    if (!gnome_url_show (new_link, NULL)){
-	    g_warning ("Unable to follow link %s\n", link);
-    }
-    
-    g_free (new_link);
-}
-/* about... */
-
 void about_command (void)
 {
     GdkPixbuf *logo;
@@ -309,13 +290,7 @@ void about_command (void)
     const char *translators = _("translator-credits");
 
     logo = gdk_pixbuf_new_from_file (PIXMAPSDIR "/gtetrinet.png", NULL);
-    
-    gtk_about_dialog_set_email_hook ((GtkAboutDialogActivateLinkFunc) handle_links,
-				     GINT_TO_POINTER (LINK_TYPE_EMAIL), NULL);
-    
-    gtk_about_dialog_set_url_hook ((GtkAboutDialogActivateLinkFunc) handle_links,
-				   GINT_TO_POINTER (LINK_TYPE_URL), NULL);
-  
+
     gtk_show_about_dialog (NULL,
 			   "name", APPNAME, 
 			   "version", APPVERSION,
diff --git a/src/config.c b/src/config.c
index e36b172..3ec8127 100644
--- a/src/config.c
+++ b/src/config.c
@@ -22,7 +22,8 @@
 #endif
 
 #include <gtk/gtk.h>
-#include <gnome.h>
+#include <gdk/gdkkeysyms.h>
+#include <glib/gi18n.h>
 #include <gconf/gconf-client.h>
 #include <stdlib.h>
 #include <string.h>
@@ -46,16 +47,16 @@ GString *currenttheme = NULL;
 extern GConfClient *gconf_client;
 
 static char *soundkeys[S_NUM] = {
-    "Sounds/Drop",
-    "Sounds/Solidify",
-    "Sounds/LineClear",
-    "Sounds/Tetris",
-    "Sounds/Rotate",
-    "Sounds/SpecialLine",
-    "Sounds/YouWin",
-    "Sounds/YouLose",
-    "Sounds/Message",
-    "Sounds/GameStart",
+    "Drop",
+    "Solidify",
+    "LineClear",
+    "Tetris",
+    "Rotate",
+    "SpecialLine",
+    "YouWin",
+    "YouLose",
+    "Message",
+    "GameStart",
 };
 
 guint defaultkeys[K_NUM] = {
@@ -82,28 +83,37 @@ void config_loadtheme (const gchar *themedir)
 {
     char buf[1024], *p;
     int i;
+    GKeyFile *keyfile;
+    GError *err = NULL;
 
-    GTET_O_STRCPY(buf, "=");
-    GTET_O_STRCAT(buf, themedir);
-    GTET_O_STRCAT(buf, "theme.cfg=/");
+    GTET_O_STRCPY(buf, themedir);
+    GTET_O_STRCAT(buf, "theme.cfg");
 
-    gnome_config_push_prefix (buf);
+    keyfile = g_key_file_new ();
+    if (!g_key_file_load_from_file (keyfile, buf, 0, &err))
+      goto bad_theme;
 
-    p = gnome_config_get_string ("Theme/Name");
+    p = g_key_file_get_string (keyfile, "Theme", "Name", &err);
     if (!p)
       goto bad_theme;
     g_free (p);
 
-    p = gnome_config_get_string ("Graphics/Blocks=blocks.png");
+    p = g_key_file_get_string (keyfile, "Graphics", "Blocks", NULL);
     if (!p)
-      goto bad_theme;
+      p = g_strdup("blocks.png");
     
     GTET_O_STRCPY(blocksfile, themedir);
     GTET_O_STRCAT(blocksfile, p);
     g_free (p);
-    bsize = gnome_config_get_int ("Graphics/BlockSize=16");
+    bsize = g_key_file_get_integer (keyfile, "Graphics", "BlockSize", &err);
+    if (err)
+    {
+      bsize = 16;
+      g_error_free (err);
+      err = NULL;
+    }
 
-    p = gnome_config_get_string ("Sounds/MidiFile");
+    p = g_key_file_get_string (keyfile, "Sounds", "MidiFile", NULL);
     if (!p)
         midifile[0] = 0;
     else
@@ -114,7 +124,7 @@ void config_loadtheme (const gchar *themedir)
     }
 
     for (i = 0; i < S_NUM; i ++) {
-        p = gnome_config_get_string (soundkeys[i]);
+        p = g_key_file_get_string (keyfile, "Sounds", soundkeys[i], NULL);
         if (p) {
             GTET_O_STRCPY(soundfiles[i], themedir);
             GTET_O_STRCAT(soundfiles[i], p);
@@ -126,7 +136,7 @@ void config_loadtheme (const gchar *themedir)
 
     sound_cache ();
 
-    gnome_config_pop_prefix ();
+    g_key_file_unref (keyfile);
 
     return;
 
@@ -141,7 +151,7 @@ void config_loadtheme (const gchar *themedir)
                                      "reverting to default."));
       gtk_dialog_run (GTK_DIALOG (mb));
       gtk_widget_destroy (mb);
-      gnome_config_pop_prefix ();
+      g_key_file_unref (keyfile);
       g_string_assign(currenttheme, DEFAULTTHEME);
       config_loadtheme (currenttheme->str);
     }
@@ -152,32 +162,44 @@ int config_getthemeinfo (char *themedir, char *name, char *author, char *desc)
 {
     char buf[1024];
     char *p = NULL;
+    GKeyFile *keyfile;
 
-    GTET_O_STRCPY (buf, "=");
-    GTET_O_STRCAT (buf, themedir);
-    GTET_O_STRCAT (buf, "theme.cfg=/");
+    GTET_O_STRCPY (buf, themedir);
+    GTET_O_STRCAT (buf, "theme.cfg");
 
-    gnome_config_push_prefix (buf);
+    keyfile = g_key_file_new ();
+    if (!g_key_file_load_from_file (keyfile, buf, 0, NULL)) {
+        g_key_file_unref (keyfile);
+        return -1;
+    }
 
-    p = gnome_config_get_string ("Theme/Name");
+    p = g_key_file_get_string (keyfile, "Theme", "Name", NULL);
     if (p == 0) {
-        gnome_config_pop_prefix ();
+        g_key_file_unref (keyfile);
         return -1;
     }
     else {
         if (name) GTET_STRCPY(name, p, 1024);
         g_free (p);
     }
-    if (author && (p = gnome_config_get_string ("Theme/Author="))) {
-        GTET_STRCPY(author, p, 1024);
-        g_free (p);
+    if (author) {
+        if ((p = g_key_file_get_string (keyfile, "Theme", "Author", NULL))) {
+            GTET_STRCPY(author, p, 1024);
+            g_free (p);
+        } else {
+            author[0] = 0;
+        }
     }
-    if (desc && (p = gnome_config_get_string ("Theme/Description="))) {
-        GTET_STRCPY(desc, p, 1024);
-        g_free (p);
+    if (desc) {
+        if ((p = g_key_file_get_string (keyfile, "Theme", "Description", NULL))) {
+            GTET_STRCPY(desc, p, 1024);
+            g_free (p);
+        } else {
+            desc[0] = 0;
+        }
     }
 
-    gnome_config_pop_prefix ();
+    g_key_file_unref (keyfile);
 
     return 0;
 }
diff --git a/src/dialogs.c b/src/dialogs.c
index 0d63305..85e8c44 100644
--- a/src/dialogs.c
+++ b/src/dialogs.c
@@ -37,6 +37,10 @@
 #include "string.h"
 #include "partyline.h"
 
+/* Adopted and renamed from libgnomeui */
+#define GTET_PAD 8
+#define GTET_PAD_SMALL 4
+
 extern GConfClient *gconf_client;
 extern GtkWidget *app;
 
@@ -161,7 +165,7 @@ void teamdialog_new (void)
     gtk_window_set_resizable (GTK_WINDOW (team_dialog), FALSE);
 
     /* entry and label */
-    hbox = gtk_hbox_new (FALSE, GNOME_PAD_SMALL);
+    hbox = gtk_hbox_new (FALSE, GTET_PAD_SMALL);
     widget = gtk_label_new (_("Team name:"));
     gtk_box_pack_start (GTK_BOX (hbox), widget ,TRUE, TRUE, 0);
     entry = gnome_entry_new ("Team");
@@ -170,7 +174,7 @@ void teamdialog_new (void)
     g_object_set (G_OBJECT (gnome_entry_gtk_entry (GNOME_ENTRY (entry))),
                   "activates_default", TRUE, NULL);
     gtk_box_pack_start (GTK_BOX (hbox), entry  ,TRUE, TRUE, 0);
-    gtk_container_set_border_width (GTK_CONTAINER (hbox), GNOME_PAD_SMALL);
+    gtk_container_set_border_width (GTK_CONTAINER (hbox), GTET_PAD_SMALL);
     gtk_box_pack_end (GTK_BOX (GTK_DIALOG (team_dialog)->vbox), hbox ,TRUE, TRUE, 0);
 
     /* pass the entry in the data pointer */
@@ -330,8 +334,8 @@ void connectdialog_new (void)
 
     /* main table */
     table1 = gtk_table_new (2, 2, FALSE);
-    gtk_table_set_row_spacings (GTK_TABLE(table1), GNOME_PAD_SMALL);
-    gtk_table_set_col_spacings (GTK_TABLE(table1), GNOME_PAD_SMALL);
+    gtk_table_set_row_spacings (GTK_TABLE(table1), GTET_PAD_SMALL);
+    gtk_table_set_col_spacings (GTK_TABLE(table1), GTET_PAD_SMALL);
 
     /* server address */
     table2 = gtk_table_new (2, 1, FALSE);
@@ -359,16 +363,16 @@ void connectdialog_new (void)
     }
     gtk_widget_show (originalradio);
     gtk_widget_show (tetrifastradio);
-    widget = gtk_hbox_new (FALSE, GNOME_PAD_SMALL);
+    widget = gtk_hbox_new (FALSE, GTET_PAD_SMALL);
     gtk_box_pack_start (GTK_BOX(widget), originalradio, 0, 0, 0);
     gtk_box_pack_start (GTK_BOX(widget), tetrifastradio, 0, 0, 0);
     gtk_widget_show (widget);
     gtk_table_attach (GTK_TABLE(table2), widget,
                       0, 1, 1, 2, GTK_FILL, GTK_FILL, 0, 0);
 
-    gtk_table_set_row_spacings (GTK_TABLE(table2), GNOME_PAD_SMALL);
-    gtk_table_set_col_spacings (GTK_TABLE(table2), GNOME_PAD_SMALL);
-    gtk_container_set_border_width (GTK_CONTAINER(table2), GNOME_PAD);
+    gtk_table_set_row_spacings (GTK_TABLE(table2), GTET_PAD_SMALL);
+    gtk_table_set_col_spacings (GTK_TABLE(table2), GTET_PAD_SMALL);
+    gtk_container_set_border_width (GTK_CONTAINER(table2), GTET_PAD);
     gtk_widget_show (table2);
     frame = gtk_frame_new (_("Server address"));
     gtk_container_add (GTK_CONTAINER(frame), table2);
@@ -396,9 +400,9 @@ void connectdialog_new (void)
     gtk_table_attach (GTK_TABLE(table2), passwordentry, 1, 2, 1, 2,
                       GTK_FILL | GTK_EXPAND, GTK_FILL | GTK_EXPAND, 0, 0);
 
-    gtk_table_set_row_spacings (GTK_TABLE(table2), GNOME_PAD_SMALL);
-    gtk_table_set_col_spacings (GTK_TABLE(table2), GNOME_PAD_SMALL);
-    gtk_container_set_border_width (GTK_CONTAINER(table2), GNOME_PAD);
+    gtk_table_set_row_spacings (GTK_TABLE(table2), GTET_PAD_SMALL);
+    gtk_table_set_col_spacings (GTK_TABLE(table2), GTET_PAD_SMALL);
+    gtk_container_set_border_width (GTK_CONTAINER(table2), GTET_PAD);
     gtk_widget_show (table2);
     frame = gtk_frame_new (_("Spectate game"));
     gtk_container_add (GTK_CONTAINER(frame), table2);
@@ -438,9 +442,9 @@ void connectdialog_new (void)
     gtk_table_attach (GTK_TABLE(table2), teamnameentry, 1, 2, 1, 2,
                       GTK_FILL | GTK_EXPAND, 0, 0, 0);
 
-    gtk_table_set_row_spacings (GTK_TABLE(table2), GNOME_PAD_SMALL);
-    gtk_table_set_col_spacings (GTK_TABLE(table2), GNOME_PAD_SMALL);
-    gtk_container_set_border_width (GTK_CONTAINER(table2), GNOME_PAD);
+    gtk_table_set_row_spacings (GTK_TABLE(table2), GTET_PAD_SMALL);
+    gtk_table_set_col_spacings (GTK_TABLE(table2), GTET_PAD_SMALL);
+    gtk_container_set_border_width (GTK_CONTAINER(table2), GTET_PAD);
     gtk_widget_show (table2);
     frame = gtk_frame_new (_("Player information"));
     gtk_container_add (GTK_CONTAINER(frame), table2);
@@ -450,7 +454,7 @@ void connectdialog_new (void)
 
     gtk_widget_show (table1);
 
-    gtk_container_set_border_width (GTK_CONTAINER (table1), GNOME_PAD_SMALL);
+    gtk_container_set_border_width (GTK_CONTAINER (table1), GTET_PAD_SMALL);
     gtk_box_pack_start (GTK_BOX (GTK_DIALOG(connectdialog)->vbox),
                         table1, TRUE, TRUE, 0);
 
@@ -490,7 +494,7 @@ gint key_dialog (char *msg)
     label = gtk_label_new (msg);
     gtk_widget_show (label);
     gtk_box_pack_start (GTK_BOX (GTK_DIALOG (dialog)->vbox),
-                        label, TRUE, TRUE, GNOME_PAD_SMALL);
+                        label, TRUE, TRUE, GTET_PAD_SMALL);
     g_signal_connect (G_OBJECT (dialog), "key-press-event",
                         GTK_SIGNAL_FUNC (key_dialog_callback), NULL);
     gtk_widget_set_events (dialog, GDK_KEY_PRESS_MASK);
@@ -777,9 +781,6 @@ void prefdialog_themelist ()
     theme_select = 0;
     gtk_list_store_clear (theme_store);
     for (i = 0; i < themecount; i ++) {
-        char *text[2];
-        text[0] = themes[i].name;
-        text[1] = 0;
         gtk_list_store_append (theme_store, &iter);
         gtk_list_store_set (theme_store, &iter, 0, themes[i].name, 1, i, -1);
         if (strcmp (themes[i].dir, currenttheme->str) == 0)
@@ -850,9 +851,9 @@ void prefdialog_new (void)
                              "Install new themes in ~/.gtetrinet/themes/"));
 
     table1 = gtk_table_new (3, 2, FALSE);
-    gtk_container_set_border_width (GTK_CONTAINER(table1), GNOME_PAD_SMALL);
+    gtk_container_set_border_width (GTK_CONTAINER(table1), GTET_PAD_SMALL);
     gtk_table_set_row_spacings (GTK_TABLE(table1), 0);
-    gtk_table_set_col_spacings (GTK_TABLE(table1), GNOME_PAD_SMALL);
+    gtk_table_set_col_spacings (GTK_TABLE(table1), GTET_PAD_SMALL);
     widget = leftlabel_new (_("Name:"));
     gtk_table_attach (GTK_TABLE(table1), widget, 0, 1, 0, 1,
                       GTK_EXPAND | GTK_FILL, 0, 0, 0);
@@ -874,21 +875,21 @@ void prefdialog_new (void)
 
     frame = gtk_frame_new (_("Selected Theme"));
     gtk_frame_set_shadow_type (GTK_FRAME(frame), GTK_SHADOW_IN);
-    gtk_container_set_border_width (GTK_CONTAINER(frame), GNOME_PAD_SMALL);
+    gtk_container_set_border_width (GTK_CONTAINER(frame), GTET_PAD_SMALL);
     gtk_widget_set_size_request (frame, 240, 100);
     gtk_container_add (GTK_CONTAINER(frame), table1);
     
     table = gtk_table_new (3, 2, FALSE);
-    gtk_container_set_border_width (GTK_CONTAINER(table), GNOME_PAD);
-    gtk_table_set_row_spacings (GTK_TABLE(table), GNOME_PAD_SMALL);
-    gtk_table_set_col_spacings (GTK_TABLE(table), GNOME_PAD_SMALL);
+    gtk_container_set_border_width (GTK_CONTAINER(table), GTET_PAD);
+    gtk_table_set_row_spacings (GTK_TABLE(table), GTET_PAD_SMALL);
+    gtk_table_set_col_spacings (GTK_TABLE(table), GTET_PAD_SMALL);
     gtk_table_attach (GTK_TABLE(table), themelist_scroll, 0, 1, 0, 3,
                       GTK_EXPAND | GTK_FILL, GTK_EXPAND | GTK_FILL, 0, 0);
     gtk_table_attach (GTK_TABLE(table), label, 1, 2, 0, 1,
                       GTK_EXPAND | GTK_FILL, GTK_FILL, 0, 0);
     gtk_table_attach (GTK_TABLE(table), frame, 1, 2, 1, 2,
                       GTK_EXPAND | GTK_FILL, GTK_EXPAND | GTK_FILL, 0, 0);
-    url = gnome_href_new ("http://gtetrinet.sourceforge.net/themes.html", _("Download new themes"));
+    url = gtk_link_button_new_with_label ("http://gtetrinet.sourceforge.net/themes.html", _("Download new themes"));
     gtk_table_attach (GTK_TABLE(table), url, 1, 2, 2, 3,
                       GTK_EXPAND | GTK_FILL, GTK_EXPAND | GTK_SHRINK, 0, 0);
     gtk_widget_show_all (table);
@@ -919,9 +920,9 @@ void prefdialog_new (void)
 		      G_CALLBACK (prefdialog_channeltoggle), NULL);
 
     table = gtk_table_new (3, 1, FALSE);
-    gtk_container_set_border_width (GTK_CONTAINER(table), GNOME_PAD);
-    gtk_table_set_row_spacings (GTK_TABLE(table), GNOME_PAD_SMALL);
-    gtk_table_set_col_spacings (GTK_TABLE(table), GNOME_PAD_SMALL);
+    gtk_container_set_border_width (GTK_CONTAINER(table), GTET_PAD);
+    gtk_table_set_row_spacings (GTK_TABLE(table), GTET_PAD_SMALL);
+    gtk_table_set_col_spacings (GTK_TABLE(table), GTET_PAD_SMALL);
     gtk_table_attach (GTK_TABLE(table), frame, 0, 1, 0, 1,
                       GTK_EXPAND | GTK_FILL, 0, 0, 0);
     gtk_widget_show (table);
@@ -963,14 +964,14 @@ void prefdialog_new (void)
     gtk_widget_show (button1);
 
     table = gtk_table_new (2, 2, FALSE);
-    gtk_container_set_border_width (GTK_CONTAINER(table), GNOME_PAD);
-    gtk_table_set_row_spacings (GTK_TABLE(table), GNOME_PAD_SMALL);
-    gtk_table_set_col_spacings (GTK_TABLE(table), GNOME_PAD_SMALL);
+    gtk_container_set_border_width (GTK_CONTAINER(table), GTET_PAD);
+    gtk_table_set_row_spacings (GTK_TABLE(table), GTET_PAD_SMALL);
+    gtk_table_set_col_spacings (GTK_TABLE(table), GTET_PAD_SMALL);
     gtk_table_attach (GTK_TABLE(table), key_scroll, 0, 1, 0, 2,
                       GTK_FILL, GTK_FILL, 0, 0);
     gtk_table_attach (GTK_TABLE(table), label, 1, 2, 0, 1,
                       GTK_FILL, 0, 0, 0);
-    frame = gtk_vbox_new (FALSE, GNOME_PAD_SMALL);
+    frame = gtk_vbox_new (FALSE, GTET_PAD_SMALL);
     gtk_box_pack_end (GTK_BOX(frame), button1, FALSE, FALSE, 0);
     gtk_box_pack_end (GTK_BOX(frame), button, FALSE, FALSE, 0);
     gtk_widget_show (frame);
@@ -1016,26 +1017,26 @@ void prefdialog_new (void)
     gtk_widget_show (button);
 
     miditable = gtk_table_new (4, 2, FALSE);
-    gtk_table_set_row_spacings (GTK_TABLE(miditable), GNOME_PAD_SMALL);
-    gtk_table_set_col_spacings (GTK_TABLE(miditable), GNOME_PAD_SMALL);
+    gtk_table_set_row_spacings (GTK_TABLE(miditable), GTET_PAD_SMALL);
+    gtk_table_set_col_spacings (GTK_TABLE(miditable), GTET_PAD_SMALL);
     gtk_table_attach (GTK_TABLE(miditable), widget, 0, 2, 0, 1,
                       GTK_EXPAND | GTK_FILL, GTK_FILL, 0, 0);
     gtk_table_attach (GTK_TABLE(miditable), midientry, 0, 2, 1, 2,
                       GTK_EXPAND | GTK_FILL, GTK_FILL, 0, 0);
     gtk_table_attach (GTK_TABLE(miditable), label, 0, 1, 2, 4,
-                      GTK_FILL, GTK_FILL, GNOME_PAD_SMALL, GNOME_PAD_SMALL);
+                      GTK_FILL, GTK_FILL, GTET_PAD_SMALL, GTET_PAD_SMALL);
     gtk_table_attach (GTK_TABLE(miditable), button, 1, 2, 2, 3,
                       GTK_EXPAND | GTK_FILL, 0, 0, 0);
     gtk_widget_show (miditable);
 
     table = gtk_table_new (3, 1, FALSE);
-    gtk_container_set_border_width (GTK_CONTAINER(table), GNOME_PAD);
-    gtk_table_set_row_spacings (GTK_TABLE(table), GNOME_PAD_SMALL);
-    gtk_table_set_col_spacings (GTK_TABLE(table), GNOME_PAD_SMALL);
+    gtk_container_set_border_width (GTK_CONTAINER(table), GTET_PAD);
+    gtk_table_set_row_spacings (GTK_TABLE(table), GTET_PAD_SMALL);
+    gtk_table_set_col_spacings (GTK_TABLE(table), GTET_PAD_SMALL);
     gtk_table_attach (GTK_TABLE(table), frame, 0, 1, 0, 1,
                       GTK_EXPAND | GTK_FILL, 0, 0, 0);
     gtk_table_attach (GTK_TABLE(table), divider, 0, 1, 1, 2,
-                      GTK_EXPAND | GTK_FILL, 0, 0, GNOME_PAD_SMALL);
+                      GTK_EXPAND | GTK_FILL, 0, 0, GTET_PAD_SMALL);
     gtk_table_attach (GTK_TABLE(table), miditable, 0, 1, 2, 3,
                       GTK_EXPAND | GTK_FILL, GTK_EXPAND | GTK_FILL, 0, 0);
     gtk_widget_show (table);
@@ -1054,7 +1055,7 @@ void prefdialog_new (void)
     gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON(soundcheck), soundenable);
     gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON(midicheck), midienable);
 
-#ifdef HAVE_ESD
+#ifdef HAVE_CANBERRAGTK
     if (midienable) prefdialog_midion ();
     else prefdialog_midioff ();
     if (soundenable) prefdialog_soundon ();
diff --git a/src/fields.c b/src/fields.c
index e0ba2f7..b4dc760 100644
--- a/src/fields.c
+++ b/src/fields.c
@@ -22,8 +22,9 @@
 #endif
 
 #include <gtk/gtk.h>
-#include <gnome.h>
+#include <glib/gi18n.h>
 #include <stdio.h>
+#include <stdlib.h>
 
 #include "gtet_config.h"
 #include "client.h"
@@ -43,7 +44,7 @@ static GtkWidget *fieldwidgets[6], *nextpiecewidget, *fieldlabels[6][6],
 
 static GtkWidget *fields_page_contents (void);
 
-static gint fields_expose_event (GtkWidget *widget, GdkEventExpose *event, int field);
+static gint fields_expose_event (GtkWidget *widget, GdkEventExpose *event, gpointer field);
 static gint fields_nextpiece_expose (GtkWidget *widget);
 static gint fields_specials_expose (GtkWidget *widget);
 
@@ -89,7 +90,7 @@ void fields_init (void)
     }
     /* we dont want the bitmap mask */
     gdk_pixbuf_render_pixmap_and_mask(pb, &blockpix, &mask, 1);
-    gdk_pixbuf_unref(pb);
+    g_object_unref(pb);
 }
 
 void fields_cleanup (void)
@@ -181,7 +182,7 @@ GtkWidget *fields_page_contents (void)
             
             /* attach the signals */
             g_signal_connect (G_OBJECT(fieldwidgets[i]), "expose_event",
-                                GTK_SIGNAL_FUNC(fields_expose_event), (gpointer)i);
+                                GTK_SIGNAL_FUNC(fields_expose_event), GINT_TO_POINTER(i));
             gtk_widget_set_events (fieldwidgets[i], GDK_EXPOSURE_MASK);
             /* set the size */
             gtk_widget_set_size_request (fieldwidgets[i],
@@ -334,11 +335,11 @@ GtkWidget *fields_page_contents (void)
 }
 
 
-gint fields_expose_event (GtkWidget *widget, GdkEventExpose *event, int field)
+gint fields_expose_event (GtkWidget *widget, GdkEventExpose *event, gpointer field)
 {
     widget = widget;
     event = event;
-    fields_refreshfield (field);
+    fields_refreshfield (GPOINTER_TO_INT (field));
     /* hide the cursor */
     if (ingame)
       gdk_window_set_cursor (widget->window, invisible_cursor);
diff --git a/src/gtetrinet.c b/src/gtetrinet.c
index 4450dd5..f91fea5 100644
--- a/src/gtetrinet.c
+++ b/src/gtetrinet.c
@@ -223,7 +223,6 @@ int main (int argc, char *argv[])
 
     /* initialise some stuff */
     fields_init ();
-    if (!g_thread_supported()) g_thread_init (NULL);
 
     /* first set up the display */
 
@@ -245,7 +244,7 @@ int main (int argc, char *argv[])
     if (icon_pixbuf)
     {
       gtk_window_set_icon (GTK_WINDOW (app), icon_pixbuf);
-      gdk_pixbuf_unref (icon_pixbuf);
+      g_object_unref (icon_pixbuf);
     }
 
     /* create the notebook */
diff --git a/src/images/Makefile.am b/src/images/Makefile.am
index 558f14a..6bcfefd 100644
--- a/src/images/Makefile.am
+++ b/src/images/Makefile.am
@@ -1,11 +1,6 @@
 EXTRA_DIST = \
-	connect.xpm \
-	disconnect.xpm \
 	fields.xpm \
 	partyline.xpm \
-	pause.xpm \
-	play.xpm \
-	stop.xpm \
 	team24.xpm \
 	winlist.xpm
 
diff --git a/src/images/connect.xpm b/src/images/connect.xpm
deleted file mode 100644
index 52640c3..0000000
--- a/src/images/connect.xpm
+++ /dev/null
@@ -1,85 +0,0 @@
-/* XPM */
-static char * connect_xpm[] = {
-"24 24 58 1",
-" 	c None",
-".	c #000000",
-"+	c #989389",
-"@	c #807D74",
-"#	c #C6C2BA",
-"$	c #34332D",
-"%	c #B7B3AA",
-"&	c #C4C2BD",
-"*	c #EAE8E3",
-"=	c #9C978D",
-"-	c #BCB9B2",
-";	c #363433",
-">	c #E2E1DD",
-",	c #F0EFEC",
-"'	c #AAA7A0",
-")	c #F0EEEB",
-"!	c #B2B0AB",
-"~	c #F9F9F8",
-"{	c #C5C3BD",
-"]	c #0F0F0D",
-"^	c #F2F0ED",
-"/	c #EBEAE6",
-"(	c #8A857B",
-"_	c #ECEBE8",
-":	c #EEECEA",
-"<	c #9F9C93",
-"[	c #F3F2F0",
-"}	c #E8E7E4",
-"|	c #E3E1DD",
-"1	c #78756B",
-"2	c #BEBBB5",
-"3	c #B3B1AA",
-"4	c #7D786E",
-"5	c #E1DFDB",
-"6	c #D1D0CC",
-"7	c #938E84",
-"8	c #C8C5BF",
-"9	c #A7A298",
-"0	c #010101",
-"a	c #8C8981",
-"b	c #A6A29B",
-"c	c #726D63",
-"d	c #CECAC3",
-"e	c #A7A49E",
-"f	c #7E7A70",
-"g	c #A09D94",
-"h	c #817D73",
-"i	c #6C685E",
-"j	c #3C3933",
-"k	c #8B877E",
-"l	c #706C62",
-"m	c #B1ADA4",
-"n	c #97938A",
-"o	c #625E54",
-"p	c #6A655B",
-"q	c #37342D",
-"r	c #646056",
-"s	c #8B877D",
-"                        ",
-"                        ",
-"                        ",
-"                        ",
-"                        ",
-"                        ",
-"          .. ..         ",
-"        ..+@.#@..       ",
-"       $%&*@.*@=-;      ",
-"      .>,'*@.)@!~{]     ",
-"......@>,'*@.)@!~{@.....",
-"*******^/(_@.:@<[}|*****",
-"@@@@@1123451.6@789@@@@@@",
-".....01abcd1.e@fghi.....",
-"      ]abcd1.e@fgh]     ",
-"      ]jklmi.n@opq.     ",
-"        ]]+r.s@..       ",
-"          ]] ].         ",
-"                        ",
-"                        ",
-"                        ",
-"                        ",
-"                        ",
-"                        "};
diff --git a/src/images/disconnect.xpm b/src/images/disconnect.xpm
deleted file mode 100644
index eecc094..0000000
--- a/src/images/disconnect.xpm
+++ /dev/null
@@ -1,69 +0,0 @@
-/* XPM */
-static char * disconnect_xpm[] = {
-"24 24 42 1",
-" 	c None",
-".	c #000000",
-"+	c #989389",
-"@	c #807D74",
-"#	c #C6C2BA",
-"$	c #43423C",
-"%	c #B7B3AA",
-"&	c #C4C2BD",
-"*	c #EAE8E3",
-"=	c #E2E1DD",
-"-	c #F0EFEC",
-";	c #AAA7A0",
-">	c #C5D2C8",
-",	c #E9EEEA",
-"'	c #F0EEEB",
-")	c #F2F0ED",
-"!	c #EBEAE6",
-"~	c #8A857B",
-"{	c #ECEBE8",
-"]	c #EEECEA",
-"^	c #78756B",
-"/	c #BEBBB5",
-"(	c #B3B1AA",
-"_	c #7D786E",
-":	c #E1DFDB",
-"<	c #D1D0CC",
-"[	c #010101",
-"}	c #8C8981",
-"|	c #A6A29B",
-"1	c #726D63",
-"2	c #CECAC3",
-"3	c #A7A49E",
-"4	c #0F0F0D",
-"5	c #F0F3F1",
-"6	c #272622",
-"7	c #8B877E",
-"8	c #706C62",
-"9	c #B1ADA4",
-"0	c #6C685E",
-"a	c #97938A",
-"b	c #646056",
-"c	c #8B877D",
-"                        ",
-"                        ",
-"                        ",
-"                        ",
-"                        ",
-"                        ",
-"         ..           ..",
-"       ..+@.         .#@",
-"      $%&*@....      .*@",
-"     .=-;*@.>>,.     .'@",
-".....@=-;*@....      .'@",
-"******)!~{@.         .]@",
-"@@@@^^/(_:^.         .<@",
-"....[^}|12^....      .3@",
-"     4}|12^.>>5.     .3@",
-"      67890....      .a@",
-"       44+b.         .c@",
-"         44           4.",
-"                        ",
-"                        ",
-"                        ",
-"                        ",
-"                        ",
-"                        "};
diff --git a/src/images/pause.xpm b/src/images/pause.xpm
deleted file mode 100644
index 26c746d..0000000
--- a/src/images/pause.xpm
+++ /dev/null
@@ -1,57 +0,0 @@
-/* XPM */
-static char * pause_xpm[] = {
-"24 24 30 1",
-" 	c None",
-".	c #000000",
-"+	c #FFFFFF",
-"@	c #737372",
-"#	c #848483",
-"$	c #888887",
-"%	c #7E7E7D",
-"&	c #A8A7A6",
-"*	c #A4A4A3",
-"=	c #A6A5A4",
-"-	c #B5B4B3",
-";	c #858483",
-">	c #AAA9A7",
-",	c #C0BFBE",
-"'	c #91908E",
-")	c #AFAEAC",
-"!	c #CACAC8",
-"~	c #A6A5A2",
-"{	c #BBBAB7",
-"]	c #D4D3D2",
-"^	c #AEADA9",
-"/	c #BFBDBA",
-"(	c #D7D5D3",
-"_	c #B2B0AD",
-":	c #C3C1BE",
-"<	c #D9D8D6",
-"[	c #B5B3B0",
-"}	c #D1CFCC",
-"|	c #DEDDDB",
-"1	c #F3F3F3",
-"                        ",
-"                        ",
-"                        ",
-"                        ",
-"                        ",
-"      .....+.....+      ",
-"      .@#$.+.@#$.+      ",
-"      .%&*.+.%&*.+      ",
-"      .%=-.+.%=-.+      ",
-"      .;>,.+.;>,.+      ",
-"      .')!.+.')!.+      ",
-"      .')!.+.')!.+      ",
-"      .')!.+.')!.+      ",
-"      .~{].+.~{].+      ",
-"      .~{].+.~{].+      ",
-"      .^/(.+.^/(.+      ",
-"      ._:<.+._:<.+      ",
-"      .[}|.+.[}|.+      ",
-"      .....+.....+      ",
-"      +++++1++++++      ",
-"                        ",
-"                        ",
-"                        ",
-"                        "};
diff --git a/src/images/play.xpm b/src/images/play.xpm
deleted file mode 100644
index 9daad31..0000000
--- a/src/images/play.xpm
+++ /dev/null
@@ -1,64 +0,0 @@
-/* XPM */
-static char * play_xpm[] = {
-"24 24 37 1",
-" 	c None",
-".	c #000000",
-"+	c #727170",
-"@	c #7B7B7A",
-"#	c #6C6B6A",
-"$	c #7F7E7D",
-"%	c #999996",
-"&	c #7D7C7B",
-"*	c #828180",
-"=	c #9D9C9A",
-"-	c #B0AFAC",
-";	c #908F8D",
-">	c #868583",
-",	c #A1A09E",
-"'	c #B4B3B1",
-")	c #B6B4B2",
-"!	c #A5A4A1",
-"~	c #898886",
-"{	c #B9B7B4",
-"]	c #BAB9B6",
-"^	c #BCBAB7",
-"/	c #C3C2BF",
-"(	c #FFFFFF",
-"_	c #8C8B89",
-":	c #A9A8A5",
-"<	c #BDBCB9",
-"[	c #C5C3C0",
-"}	c #D0CECC",
-"|	c #8F8E8C",
-"1	c #B5B3B1",
-"2	c #C7C6C3",
-"3	c #D2D1CE",
-"4	c #B9B6B4",
-"5	c #D8D7D5",
-"6	c #A19F9D",
-"7	c #C7C6C4",
-"8	c #A7A6A3",
-"                        ",
-"                        ",
-"                        ",
-"                        ",
-"         .              ",
-"         ..             ",
-"         .+.            ",
-"         .@#.           ",
-"         .$%&.          ",
-"         .*=-;.         ",
-"         .>,')!.        ",
-"         .~!{]^/.(      ",
-"         ._:<[}.(       ",
-"         .|123.(        ",
-"         .=45.(         ",
-"         .67.(          ",
-"         .8.(           ",
-"         ..(            ",
-"         .(             ",
-"         (              ",
-"                        ",
-"                        ",
-"                        ",
-"                        "};
diff --git a/src/images/stop.xpm b/src/images/stop.xpm
deleted file mode 100644
index d4e078b..0000000
--- a/src/images/stop.xpm
+++ /dev/null
@@ -1,56 +0,0 @@
-/* XPM */
-static char * stop_xpm[] = {
-"24 24 29 1",
-" 	c None",
-".	c #000000",
-"+	c #FFFFFF",
-"@	c #737372",
-"#	c #848483",
-"$	c #888887",
-"%	c #7E7E7D",
-"&	c #A8A7A6",
-"*	c #A4A4A3",
-"=	c #A6A5A4",
-"-	c #B5B4B3",
-";	c #858483",
-">	c #AAA9A7",
-",	c #C0BFBE",
-"'	c #91908E",
-")	c #AFAEAC",
-"!	c #CACAC8",
-"~	c #A6A5A2",
-"{	c #BBBAB7",
-"]	c #D4D3D2",
-"^	c #AEADA9",
-"/	c #BFBDBA",
-"(	c #D7D5D3",
-"_	c #B2B0AD",
-":	c #C3C1BE",
-"<	c #D9D8D6",
-"[	c #B5B3B0",
-"}	c #D1CFCC",
-"|	c #DEDDDB",
-"                        ",
-"                        ",
-"                        ",
-"                        ",
-"                        ",
-"     .............+     ",
-"     .@#$$$$$$$$$.+     ",
-"     .%&*********.+     ",
-"     .%=---------.+     ",
-"     .;>,,,,,,,,,.+     ",
-"     .')!!!!!!!!!.+     ",
-"     .')!!!!!!!!!.+     ",
-"     .~{]]]]]]]]].+     ",
-"     .~{]]]]]]]]].+     ",
-"     .^/(((((((((.+     ",
-"     ._:<<<<<<<<<.+     ",
-"     .[}|||||||||.+     ",
-"     .............+     ",
-"     ++++++++++++++     ",
-"                        ",
-"                        ",
-"                        ",
-"                        ",
-"                        "};
diff --git a/src/misc.c b/src/misc.c
index 4a48f3c..6ac3451 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -18,7 +18,6 @@
  */
 
 #include <gtk/gtk.h>
-#include <gnome.h>
 #include <unistd.h>
 #include <stdlib.h>
 
@@ -129,7 +128,7 @@ void textbox_setup (void)
     tag_table = gtk_text_buffer_get_tag_table(buffer);
 }
 
-void textbox_addtext (GtkTextView *textbox, const unsigned char *str)
+void textbox_addtext (GtkTextView *textbox, const char *str)
 {
     /* At this point, str should be valid utf-8 except for some 0xFF-bytes for
      * formatting. */
@@ -143,8 +142,8 @@ void textbox_addtext (GtkTextView *textbox, const unsigned char *str)
     gboolean attr_underline = FALSE;
     /* GtkAdjustment *textboxadj = GTK_TEXT_VIEW(textbox)->vadjustment; */
     GtkTextIter iter;
-    guchar* p;
-    guchar* text=g_strdup(str);
+    gchar* p;
+    gchar* text=g_strdup(str);
 
                 
     /* is the scroll bar at the bottom ?? */
@@ -165,8 +164,10 @@ void textbox_addtext (GtkTextView *textbox, const unsigned char *str)
         gtk_text_buffer_insert (textbox->buffer, &iter, "\n", 1);
 
     /* For-loop with utf-8 support. - vidar*/
+    /* XXX: Relies on g_utf8_next_char advancing by one byte on an invalid start byte,
+       which is undefined by the documentation of g_utf8_next_char. -stump */
     for(p=text; p[0]; p=g_utf8_next_char(p)) {
-        tmp=p[0];  /* Only for checking color codes now.*/
+        tmp=(guchar)p[0];  /* Only for checking color codes now.*/
         
 	if (tmp == TETRI_TB_RESET) {
 	    lastcolor = color = gtet_text_tags[0].t_c;
@@ -303,7 +304,7 @@ char *nocolor (char *str)
 
   g_string_assign(ret, str);
 
-  p = scan = ret->str;
+  p = scan = (signed char *)ret->str;
   while (*scan != 0)
   {
     if ((*scan > 0x1F) || (*scan < 0x0)) *p++ = *scan;
diff --git a/src/misc.h b/src/misc.h
index 1534cc6..6d6df88 100644
--- a/src/misc.h
+++ b/src/misc.h
@@ -7,7 +7,7 @@ extern void leftlabel_set (GtkWidget *align, char *str);
 extern int randomnum (int n);
 extern void fdreadline (int fd, char *buf);
 extern void textbox_setup (void);
-extern void textbox_addtext (GtkTextView *textbox, const unsigned char *text);
+extern void textbox_addtext (GtkTextView *textbox, const char *text);
 extern void adjust_bottom_text_view (GtkTextView *);
 extern char *nocolor (char *str);
 
diff --git a/src/partyline.c b/src/partyline.c
index 16d835f..a013611 100644
--- a/src/partyline.c
+++ b/src/partyline.c
@@ -22,7 +22,8 @@
 #endif
 
 #include <gtk/gtk.h>
-#include <gnome.h>
+#include <gdk/gdkkeysyms.h>
+#include <glib/gi18n.h>
 #include <string.h>
 #include <stdio.h>
 #include <stdarg.h>
@@ -31,7 +32,6 @@
 #include "tetrinet.h"
 #include "partyline.h"
 #include "misc.h"
-#include "commands.h"
 
 int timestampsenable;
 gboolean list_enabled;
diff --git a/src/sound.c b/src/sound.c
index 93e4797..1d5c2dc 100644
--- a/src/sound.c
+++ b/src/sound.c
@@ -22,10 +22,10 @@
 #endif
 
 #include <gtk/gtk.h>
-#include <gnome.h>
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <signal.h>
+#include <stdlib.h>
 #include "sound.h"
 
 extern char **environ;
@@ -36,12 +36,11 @@ char soundfiles[S_NUM][1024];
 char midifile[1024];
 char midicmd[1024];
 
-#ifdef HAVE_ESD
+#ifdef HAVE_CANBERRAGTK
 
-#include <esd.h>
-#include <libgnome/gnome-sound.h>
+#include <canberra-gtk.h>
 
-static int soundsamples[S_NUM];
+static char *soundsamples[S_NUM] = {NULL};
 static int midipid = 0;
 
 void sound_cache (void)
@@ -49,20 +48,26 @@ void sound_cache (void)
     int i;
     if (!soundenable) return;
     for (i = 0; i < S_NUM; i ++) {
-        if (soundsamples[i])
-            esd_sample_free (gnome_sound_connection_get (), soundsamples[i]);
-        if (soundfiles[i][0])
-            soundsamples[i] = gnome_sound_sample_load (soundfiles[i], soundfiles[i]);
-        else
-            soundsamples[i] = 0;
+        if (soundsamples[i] != NULL)
+            g_free (soundsamples[i]);
+        if (soundfiles[i][0]) {
+            ca_context_cache (ca_gtk_context_get (), CA_PROP_MEDIA_FILENAME, soundfiles[i], NULL);
+            soundsamples[i] = g_strdup (soundfiles[i]);
+        } else {
+            soundsamples[i] = NULL;
+        }
     }
 }
 
 void sound_playsound (int id)
 {
     if (!soundenable) return;
-    if (soundsamples[id] > 0)
-      esd_sample_play (gnome_sound_connection_get (), soundsamples[id]);
+    if (soundsamples[id] != NULL)
+      ca_context_play (ca_gtk_context_get (), 0,
+                         CA_PROP_MEDIA_FILENAME, soundsamples[id],
+                         /* If GTK says sounds are disabled, override it. */
+                         CA_PROP_CANBERRA_ENABLE, "1",
+                         NULL);
 }
 
 void sound_playmidi (char *file)
diff --git a/src/tetrinet.c b/src/tetrinet.c
index ca5c5cf..3cb8582 100644
--- a/src/tetrinet.c
+++ b/src/tetrinet.c
@@ -75,7 +75,7 @@ int specialblocknum = 0;
 /*
  * this will have the number of /list commands sended and waiting for answer
  */
-gint list_issued;
+int list_issued;
 
 FIELD fields[7];
 
diff --git a/src/tetrinet.h b/src/tetrinet.h
index af162b5..40bf82e 100644
--- a/src/tetrinet.h
+++ b/src/tetrinet.h
@@ -1,5 +1,3 @@
-#include <gnome.h>
-
 #define FIELDWIDTH 12
 #define FIELDHEIGHT 22
 
@@ -17,7 +15,7 @@ extern int gmsgstate;
 
 extern char specialblocks[256];
 extern int specialblocknum;
-extern gint list_issued;
+extern int list_issued;
 
 extern void tetrinet_inmessage (enum inmsg_type msgtype, char *data);
 extern void tetrinet_playerline (const char *text);
diff --git a/src/winlist.c b/src/winlist.c
index 4ac8c1e..5e4ff5b 100644
--- a/src/winlist.c
+++ b/src/winlist.c
@@ -22,7 +22,7 @@
 #endif
 
 #include <gtk/gtk.h>
-#include <gnome.h>
+#include <glib/gi18n.h>
 #include <string.h>
 #include <stdio.h>
 
@@ -45,11 +45,11 @@ GtkWidget *winlist_page_new (void)
     /* Load the icons and scale them */
     pixbuf = gdk_pixbuf_new_from_file (GTETPIXMAPSDIR "/team.png",  NULL);
     team_icon = gdk_pixbuf_scale_simple (pixbuf, 24, 24, GDK_INTERP_BILINEAR);
-    gdk_pixbuf_unref (pixbuf);
+    g_object_unref (pixbuf);
     
     pixbuf = gdk_pixbuf_new_from_file (GTETPIXMAPSDIR "/alone.png", NULL);
     alone_icon = gdk_pixbuf_scale_simple (pixbuf, 24, 24, GDK_INTERP_BILINEAR);
-    gdk_pixbuf_unref (pixbuf);
+    g_object_unref (pixbuf);
 
     winlist = gtk_tree_view_new_with_model (GTK_TREE_MODEL (winlist_store));
 
